cumulus/client/cirrus-executor/src/aux_schema.rs

```diff
@@ -50,31 +50,33 @@ pub(super) fn write_execution_receipt<Backend: AuxStore, Block: BlockT, PBlock:
 
 	let mut new_first_saved_receipt = first_saved_receipt;
 
-	let keys_to_delete =
-		if best_execution_chain_number >= first_saved_receipt + PRUNING_DEPTH.saturated_into() {
-			new_first_saved_receipt =
-				best_execution_chain_number.saturating_sub((PRUNING_DEPTH - 1).saturated_into());
-
-			let mut keys_to_delete = vec![];
-			let mut to_delete_start = first_saved_receipt;
-			while to_delete_start < new_first_saved_receipt {
-				let delete_block_number_key =
-					(EXECUTION_RECEIPT_BLOCK_NUMBER, to_delete_start).encode();
-				if let Some(hashes_to_delete) =
-					load_decode::<_, Vec<Block::Hash>>(backend, delete_block_number_key.as_slice())?
-				{
-					keys_to_delete.extend(
-						hashes_to_delete.into_iter().map(|h| (EXECUTION_RECEIPT_KEY, h).encode()),
-					);
-					keys_to_delete.push(delete_block_number_key);
-				}
-				to_delete_start = to_delete_start.saturating_add(One::one());
+	let keys_to_delete = if best_execution_chain_number.saturating_sub(first_saved_receipt) >=
+		PRUNING_DEPTH.into()
+	{
+		// `first_saved_receipt` starts from 1 instead of 0, hence `PRUNING_DEPTH` - 1.
```

Comment:
I don't understand why it matters which number starts with. We are just subtracting a number from another.

Reply:
Hmm, still hard to understand with this supplementary example https://github.com/subspace/subspace/pull/490#discussion_r879996823 ?